robottelo/ui/session.py

```diff
@@ -1,55 +1,245 @@
+from robottelo.ui.computeprofile import ComputeProfile
+from robottelo.ui.computeresource import ComputeResource
+from robottelo.ui.configgroups import ConfigGroups
+from robottelo.ui.container import Container
+from robottelo.ui.contenthost import ContentHost
+from robottelo.ui.contentviews import ContentViews
+from robottelo.ui.dashboard import Dashboard
+from robottelo.ui.discoveredhosts import DiscoveredHosts
+from robottelo.ui.discoveryrules import DiscoveryRules
+from robottelo.ui.dockertag import DockerTag
+from robottelo.ui.domain import Domain
+from robottelo.ui.environment import Environment
+from robottelo.ui.errata import Errata
+from robottelo.ui.gpgkey import GPGKey
+from robottelo.ui.hardwaremodel import HardwareModel
+from robottelo.ui.hostcollection import HostCollection
+from robottelo.ui.hostgroup import Hostgroup
+from robottelo.ui.hosts import Hosts
+from robottelo.ui.job import Job
+from robottelo.ui.job_template import JobTemplate
+from robottelo.ui.ldapauthsource import LdapAuthSource
+from robottelo.ui.lifecycleenvironment import LifecycleEnvironment
+from robottelo.ui.location import Location
 from robottelo.ui.login import Login
+from robottelo.ui.medium import Medium
+from robottelo.ui.my_account import MyAccount
 from robottelo.ui.navigator import Navigator
+from robottelo.ui.operatingsys import OperatingSys
+from robottelo.ui.org import Org
+from robottelo.ui.oscapcontent import OpenScapContent
+from robottelo.ui.oscappolicy import OpenScapPolicy
+from robottelo.ui.oscapreports import OpenScapReports
+from robottelo.ui.packages import Package
+from robottelo.ui.partitiontable import PartitionTable
+from robottelo.ui.products import Products
+from robottelo.ui.puppetclasses import PuppetClasses
+from robottelo.ui.puppetmodule import PuppetModule
+from robottelo.ui.registry import Registry
+from robottelo.ui.repository import Repos
+from robottelo.ui.rhai import RHAI
+from robottelo.ui.role import Role
+from robottelo.ui.settings import Settings
+from robottelo.ui.scparams import SmartClassParameter
+from robottelo.ui.smart_variable import SmartVariable
+from robottelo.ui.subnet import Subnet
+from robottelo.ui.subscription import Subscriptions
+from robottelo.ui.sync import Sync
+from robottelo.ui.syncplan import Syncplan
+from robottelo.ui.task import Task
+from robottelo.ui.template import Template
+from robottelo.ui.trend import Trend
+from robottelo.ui.usergroup import UserGroup
+from robottelo.ui.user import User
 
 _org_cache = {}
+LOGGER = logging.getLogger(__name__)
 
 
 class Session(object):
     """A session context manager that manages login and logout"""
 
-    def __init__(self, browser, user=None, password=None):
-        self._login = Login(browser)
-        self.browser = browser
-        self.nav = Navigator(browser)
-        self.password = password
-        self.user = user
+    def __init__(self, test, user=None, password=None):
+        self.test = test
```

Comment:
@renzon in your example `session` is taken via `pytest fixture` right? in that example how the `test_id` is going to be passed to session object?

```
@tier1
def test_positive_search_autocomplete(session, local_ui_org, ui_org_manager):
    """Search for an organization can be auto-completed by partial
    name
    :id: f3c492ab-46fb-4b1d-b5d5-29a82385d681
    :expectedresults: Auto search for created organization works as
        intended
    :CaseImportance: Critical
    """
    part_string = local_ui_org.name[:3]
    auto_search = ui_org_manager.auto_complete_search(
        session.nav.go_to_org,
        locators['org.org_name'], part_string, local_ui_org.name,
        search_key='name')
    assert auto_search is not None
```